### PR TITLE
Re-include astropy/*/src/*.c in .gitignore

### PR DESCRIPTION
**Pull request by taldcroft**
### Description

The blanket exclusion of `*.c` in `.gitignore` is inconvenient when working with C code. This prevents my editor from quickly finding the file or including it in search, etc.  I think that `astropy/*/src/*.c` is fine, but maybe we could just allow everything in `astropy/*/src/*`?
```
✗ ls astropy/*/src/   
astropy/convolution/src/:
convolve.c  convolve.h

astropy/modeling/src/:
projections.c        projections.c.templ  wcsconfig.h

astropy/utils/src/:
compiler.c

astropy/wcs/src/:
astropy_wcs.c         pyutil.c              wcslib_auxprm_wrap.c
astropy_wcs_api.c     sip.c                 wcslib_tabprm_wrap.c*
distortion.c          sip_wrap.c            wcslib_wrap.c*
distortion_wrap.c     str_list_proxy.c      wcslib_wtbarr_wrap.c
docstrings.c          unit_list_proxy.c
pipeline.c            util.c
```

### CODE REVIEW

**Comment by pllim** [comment]
Hello. I converted your PR to a draft PR due to #10706 .

File: .gitignore [hunk 1]
```diff
@@ -8,8 +8,10 @@
 __pycache__
 
 # Ignore .c files by default to avoid including generated code. If you want to
-# add a non-generated .c extension, use `git add -f filename.c`.
+# add a non-generated .c extension, put that into the src/ subdirectory of a
+# package or else use `git add -f filename.c`.
 *.c
+! astropy/*/src/*.c
```
Comment (pllim) [inline .gitignore:14]: Fine by me. Just not sure about the whitespace. The examples in https://www.atlassian.com/git/tutorials/saving-changes/gitignore don't have whitespace.

```suggestion
!astropy/*/src/*.c
```

**Comment by taldcroft** [comment]
@pllim - I committed the change you suggested.

**Review by pllim — APPROVED** [review]
Failures are unrelated.

**Comment by pllim** [comment]
Thanks!

**Pull request merged**